### ISSUE TITLE
Manually adding a group via group field does not work anymore

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#7180]
JabRef build today.

- [x] **Mandatory**: I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

See https://github.com/JabRef/jabref/issues/6690#issuecomment-668440560

Alternatively, manually edit the bib entry

![grafik](https://user-images.githubusercontent.com/1366654/101996322-c2dc9700-3cd1-11eb-885c-aa885e2419c9.png)

Idea: Maybe JabRef's functionality regarding groups changed from August to now --> JabRef seems to parse groups defined at the lower part of JabRef and does not automatically create the groups:

```bibtex
@Comment{jabref-meta: grouping:
0 AllEntriesGroup:;
1 StaticGroup:By status\;2\;1\;0x8a8a8aff\;\;\;;
2 StaticGroup:Last read\;0\;0\;0x80334dff\;book\;\;;
2 StaticGroup:Skimmed\;0\;0\;0xe6994dff\;eye\;\;;
2 StaticGroup:To be read\;0\;0\;0x336633ff\;bullseye\;\;;
1 StaticGroup:By rating\;0\;1\;0x808080ff\;\;\;;
1 StaticGroup:For Project X\;0\;1\;0x336633ff\;security\;\;;
1 StaticGroup:For teaching\;0\;1\;0x4d3399ff\;school\;\;;
1 StaticGroup:My papers\;0\;1\;0x8a8a8aff\;note\;\;;
}
```

**Comment by k3KAW8Pnf7mkmdSMPHz27** [comment]
What do you mean by "manually adding a group via group field"? Is there a way of doing this without editing the bib entry?

Should creating a group by editing a bib entry work? I am thinking that there can be some confusion regarding empty ExplicitGroups (the ones with manually added entries)? Should they remain in the group list or be removed? What happens if you make a typo (i.e., will it create several empty groups)?

**Comment by Siedlerchr** [comment]
Adding a new group by adding a new group in the entry's group field. Does not make any sense toe me

**Comment by k3KAW8Pnf7mkmdSMPHz27** [comment]
I just don't know what this issue refers too. Following the steps in https://github.com/JabRef/jabref/issues/6690#issuecomment-668440560 I get the result I expect ("For fun" is empty)?

If the issue is that it is not possible to change the name in the .bib file manually, that should have been the case for quite a while, but the old format should still be supported? I suppose an alternative would be to generate a unique ID for groups and store the name separately (resolve JabRef strings in groups! 😁 ). There has been some previous discussion on this topic, I can drag up the references if this is what this issue is about?

**Comment by koppor** [comment]
> I am thinking that there can be some confusion regarding empty ExplicitGroups (the ones with manually added entries)? Should they remain in the group list or be removed? What happens if you make a typo (i.e., will it create several empty groups)?

I see your point. I would say that deletion of "wrong" empty static groups is simpler than creating explict groups by hand. I am more a guy editing the BibTeX source than using some UI. Since JabRef perfectly shows the BibTeX source, I just edit that. I can add new authors, keywords, ... and all grouping works (hopefully ^^). Why not at explicit groups?


... some more thoughts ...

> What do you mean by "manually adding a group via group field"? Is there a way of doing this without editing the bib entry?

Using drag'n'drop would IMHO be one way.

I actually solved the issue by replacing ALL of the content of the file as described. Thereby, I also replaced

    1 StaticGroup:For teaching\;0\;1\;0x4d3399ff\;school\;\;;

by

    1 StaticGroup:For fun\;0\;1\;0x4d3399ff\;school\;\;;


And then, the group was shown.

> Should creating a group by editing a bib entry work?

I was expecting so. I thought, in 2016 we changed the groups format. See https://github.com/JabRef/jabref/pull/1276, https://github.com/JabRef/jabref/blob/v3.8.2/CHANGELOG.md#changed-6, and a small announcement at http://blog.jabref.org/#october-04-2017-%E2%80%93-jabref-4-0. 

We explicitly moved the groups into the `groups` field. Similar that I can change the author name in the BibTeX source and the automatic groups change accordingly, I am expecting, assigning an explicit group "magically" creates a static group.
